File: app/src/main/java/com/pandatone/kumiwake/sekigime/SelectTableType.java
package com.pandatone.kumiwake.sekigime;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;

import com.pandatone.kumiwake.R;
import com.pandatone.kumiwake.customize.CustomDialogSekigime;

import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by atsushi_2 on 2016/07/13.
 */
public class SelectTableType extends AppCompatActivity {

    LayoutInflater inflater;
    String title="";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.select_sekigime_type);
        ButterKnife.bind(this);
        inflater = (LayoutInflater) getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @OnClick(R.id.square_table)
    void onSquareClicked() {
        DrawTableView.tableType = "square";
        title = getString(R.string.square_table);
        ConfirmationDialog(title,1);
    }

    @OnClick(R.id.parallel_table)
    void onParallelClicked() {
        DrawTableView.tableType = "parallel";
        title = getString(R.string.parallel_table);
        ConfirmationDialog(title,2);
    }

    @OnClick(R.id.circle_table)
    void onCircleClicked() {
        DrawTableView.tableType = "circle";
        title = getString(R.string.circle_table);
        ConfirmationDialog(title,3);
    }

    @OnClick(R.id.counter_table)
    void onCounterClicked() {
        DrawTableView.tableType = "counter";
        title = getString(R.string.counter_table);
        ConfirmationDialog(title,4);
    }

    public void ConfirmationDialog(String title, int position) {
        CustomDialogSekigime customDialog = new CustomDialogSekigime();
        customDialog.setTitle(title);
        customDialog.setPosition(position);
        customDialog.show(getFragmentManager(), "Btn");
    }
}

File: app/src/main/java/com/pandatone/kumiwake/sekigime/SelectTableType.kt
package com.pandatone.kumiwake.sekigime

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.Button
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity

import com.pandatone.kumiwake.R
import com.pandatone.kumiwake.sekigime.function.CustomDialogSekigime
import com.pandatone.kumiwake.sekigime.function.DrawTableView

/**
 * Created by atsushi_2 on 2016/07/13.
 */
class SelectTableType : AppCompatActivity() {

    private lateinit var inflater: LayoutInflater
    private var title = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.select_sekigime_type)

        inflater = getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        findViewById<ImageButton>(R.id.square_table).setOnClickListener { onSquareClicked() }
        findViewById<ImageButton>(R.id.parallel_table).setOnClickListener { onParallelClicked() }
        findViewById<ImageButton>(R.id.circle_table).setOnClickListener { onCircleClicked() }
        findViewById<ImageButton>(R.id.counter_table).setOnClickListener { onCounterClicked() }
    }

    private fun onSquareClicked() {
        DrawTableView.tableType = "square"
        title = getString(R.string.square_table)
        confirmationDialog(title, 1)
    }

    private fun onParallelClicked() {
        DrawTableView.tableType = "parallel"
        title = getString(R.string.parallel_table)
        confirmationDialog(title, 2)
    }

    private fun onCircleClicked() {
        DrawTableView.tableType = "circle"
        title = getString(R.string.circle_table)
        confirmationDialog(title, 3)
    }

    private fun onCounterClicked() {
        DrawTableView.tableType = "counter"
        title = getString(R.string.counter_table)
        confirmationDialog(title, 4)
    }

    private fun confirmationDialog(title: String, position: Int) {
        val customDialog = CustomDialogSekigime()
        customDialog.setTitle(title)
        customDialog.setPosition(position)
        customDialog.show(supportFragmentManager, "Btn")
    }
}
